File: weather/src/main/java/com/nerdery/pkarels/weather/data/model/ForecastCondition.java
package com.nerdery.pkarels.weather.data.model;

import com.google.gson.annotations.SerializedName;

import java.util.Date;

/**
 * Specific weather condition for time and location
 */
public class ForecastCondition {
    String summary;
    String icon;
    @SerializedName("temperature")
    double temp;
    long time;

    private boolean isLowest;
    private boolean isHightest;

    public ForecastCondition(String summary, String icon, double temp, long time) {
        this.summary = summary;
        this.icon = icon;
        this.temp = temp;
        this.time = time;
    }

    /**
     * Text summary of weather condition
     * @return Summary
     */
    public String getSummary() {
        return summary;
    }

    /**
     * Icon name of weather condition
     * @return Icon Name
     */
    public String getIcon() {
        return icon;
    }

    /**
     * Temperature in degrees of {@link TempUnit} sent during request
     * @return Temperature
     */
    public long getTemp() {
        return Math.round(temp);
    }

    /**
     * Time/Date of Forecast Condition
     * @return Date
     */
    public Date getTime() {
        return new Date(time);
    }

    public boolean isLowest() {
        return isLowest;
    }

    public void setLowest(boolean lowest) {
        isLowest = lowest;
    }

    public boolean isHightest() {
        return isHightest;
    }

    public void setHightest(boolean hightest) {
        isHightest = hightest;
    }
}

File: weather/src/main/java/com/nerdery/pkarels/weather/model/ForecastCondition.kt
package com.nerdery.pkarels.weather.model

import com.google.gson.annotations.SerializedName
import com.nerdery.pkarels.life.TempUnit
import com.nerdery.pkarels.life.Util
import org.threeten.bp.LocalDateTime
import org.threeten.bp.ZoneOffset
import java.util.*

/**
 * Specific weather condition for time and location
 */
data class ForecastCondition(val summary: String,
                             val icon: String,
                             @field:SerializedName("temperature")
                        val temp: Double,
                             private val time: Long) {

    var isLowest: Boolean = false
    var isHighest: Boolean = false

    var tempUnit = TempUnit.FAHRENHEIT

    // Todo - move to ViewModel
    fun getTemperature(): String {
        return Util.round(temp, 1)
    }

    // TODO: move to ViewModel
    fun getTimeInMillis(): Long {
        return time * 1000
    }

    /**
     * Time/Date of Forecast Condition
     * @return Date
     */
    fun getTimeAsDate(): Date {
        return Date(time * 1000)
    }

    fun getTimeAsLocalDateTime(): LocalDateTime {
        return LocalDateTime.ofEpochSecond(time, 0, ZoneOffset.UTC)
    }
}
